spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesMapper.java

```diff
@@ -71,13 +73,23 @@ void customizeAdminBuilder(PulsarAdminBuilder adminBuilder, PulsarConnectionDeta
 
 	private void customizeAuthentication(AuthenticationConsumer authentication,
 			PulsarProperties.Authentication properties) {
-		if (StringUtils.hasText(properties.getPluginClassName())) {
+		if (!StringUtils.hasText(properties.getPluginClassName())) {
+			return;
+		}
+		try {
+			// sort keys for testing and readability
+			Map<String, String> params = new TreeMap<>(properties.getParam());
+			String authParamString;
 			try {
-				authentication.accept(properties.getPluginClassName(), properties.getParam());
```

Comment:
@philwebb it looks like going w/ the map of auth params was a short-lived avenue as the use of auth params map has been deprecated in favor of the auth params string. I think we should keep the properties using the map approach and simply json encode them to the expected auth param string (implementation detail).